test/controllers/journeys/income/IncomeNotCountedAsTurnoverControllerSpec.scala

```diff
@@ -40,122 +41,167 @@ class IncomeNotCountedAsTurnoverControllerSpec extends SpecBase with MockitoSuga
 
-        val result = route(application, request).value
+      userScenarios.foreach { userScenario =>
+        s"when language is ${isWelshToString(userScenario.isWelsh)} and user is an ${isAgentToString(userScenario.isAgent)}" - {
 
-        val view = application.injector.instanceOf[IncomeNotCountedAsTurnoverView]
+          "must return OK and the correct view for a GET" in {
 
-        status(result) mustEqual OK
-        contentAsString(result) mustEqual view(form, NormalMode, taxYear)(request, messages(application)).toString
-      }
-    }
+            val application          = applicationBuilder(userAnswers = Some(emptyUserAnswers), userScenario.isAgent).build()
+            implicit val messagesApi = application.injector.instanceOf[MessagesApi]
 
-    "must populate the view correctly on a GET when the question has previously been answered" in {
+            running(application) {
+              val request = buildRequest(GET, incomeNotCountedAsTurnoverRoute(false, NormalMode), userScenario.isAgent)
 
-      val userAnswers = UserAnswers(userAnswersId).set(IncomeNotCountedAsTurnoverPage, true).success.value
+              val result = route(application, request).value
 
-      val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
+              val langResult = if (userScenario.isWelsh) result.map(_.withLang(cyLang)) else result
 
-      running(application) {
-        val request = FakeRequest(GET, incomeNotCountedAsTurnoverRoute)
+              val view = application.injector.instanceOf[IncomeNotCountedAsTurnoverView]
 
-        val view = application.injector.instanceOf[IncomeNotCountedAsTurnoverView]
+              val expectedResult =
+                view(userScenario.form, NormalMode, isAgentToString(userScenario.isAgent), taxYear)(
+                  request,
+                  messages(application, userScenario.isWelsh)).toString
 
-        val result = route(application, request).value
+              status(result) mustEqual OK
+              contentAsString(langResult) mustEqual expectedResult
+            }
+          }
 
-        status(result) mustEqual OK
-        contentAsString(result) mustEqual view(form.fill(true), NormalMode, taxYear)(request, messages(application)).toString
-      }
-    }
+          "must populate the view correctly on a GET when the question has previously been answered" in {
+
+            val userAnswers = UserAnswers(userAnswersId).set(IncomeNotCountedAsTurnoverPage, true).success.value
 
-    "must redirect to the next page when valid data is submitted" in {
+            val application          = applicationBuilder(userAnswers = Some(userAnswers), userScenario.isAgent).build()
+            implicit val messagesApi = application.injector.instanceOf[MessagesApi]
 
-      val mockSessionRepository = mock[SessionRepository]
+            running(application) {
+              val request = FakeRequest(GET, incomeNotCountedAsTurnoverRoute(false, CheckMode))
```

Comment:
Why do we use `CheckMode` in the test here?

Reply:
Because there are already userAnswers for the page so this is editing the page answers - is a check not a normal route.